subsystems/networking/scanning.go

```diff
@@ -28,21 +28,31 @@ func (n *Networking) networkScan(ctx context.Context) error {
 		return errw.Errorf("cannot find hotspot interface: %s", n.Config().HotspotInterface)
 	}
 
+	state, reason, err := wifiDev.GetPropertyStateReason()
+	if err != nil {
+		return errw.Wrap(err, "getting wifi state and reason")
+	}
+
+	if state != gnm.NmDeviceStateDisconnected && state != gnm.NmDeviceStateActivated {
+		n.logger.Debugf("wifi device state: %s, reason: %s, skipping scan", state, reason)
+		return nil
+	}
+
```

Comment:
This should help reduce the flood of "Scanning not allowed while unavailable" messages Eliot and Bill reported.